controllers/configs.go

```diff
@@ -48,6 +48,17 @@ func (r *TalosControlPlaneReconciler) talosconfigForMachines(ctx context.Context
 	}
 
 	clusterName := tcp.GetLabels()["cluster.x-k8s.io/cluster-name"]
+	if clusterName == "" {
```

Comment:
this looks like something which is already in cluster-api:

```go
// GetOwnerCluster returns the Cluster object owning the current resource.
func GetOwnerCluster(ctx context.Context, c client.Client, obj metav1.ObjectMeta) (*clusterv1.Cluster, error) {
```

Reply:
it doesn't seem equivalent to me. In our code we read the cluster name from the label.

Reply:
anyways we should use proper way - see that function. not sure if we need our own ways vs. what Cluster API does

Reply:
gotcha, fixed